packages/react-dom/src/__tests__/ReactCompositeComponentState-test.js

```diff
@@ -445,20 +506,29 @@ describe('ReactCompositeComponent-state', () => {
 
     // Mount
     const container = document.createElement('div');
-    expect(() => ReactDOM.render(<Test />, container)).toErrorDev(
+    const root = ReactDOMClient.createRoot(container);
+    expect(() => {
+      ReactDOM.flushSync(() => {
+        root.render(<Test />);
+      });
+    }).toErrorDev(
       'Warning: Test.componentWillMount(): Assigning directly to ' +
         "this.state is deprecated (except inside a component's constructor). " +
         'Use setState instead.',
     );
 
-    expect(ops).toEqual([
+    assertLog([
+      'render -- step: 3, extra: false',
+      'callback -- step: 3, extra: false',
+
+      // A second time for the retry.
```

Comment:
`createRoot` automatically retries when we error in render.

Reply:
Maybe add this context to the comment on this line